music/src/core/visualizer.ts

```diff
@@ -456,3 +458,205 @@ export class PianoRollSVGVisualizer extends BaseVisualizer {
+ */
+export enum ScrollType {
+  PAGE = 0,
+  NOTE = 1,
+  BAR = 2
+}
+
+/**
+ * An interface for providing extra configurable properties to a Visualizer
+ * extending the basic configurable properties of `VisualizerConfig`.
+ *
+ * @param defaultKey The musical key the score must use to adapt the score to
+ * the right accidentals. It can be overwritten with
+ * `NoteSequence.keySignatures` value at time or step 0. If not assigned it
+ * will be asumed C key.
+ * @param instruments The subset of the `NoteSequence` instrument track numbers
+ * which should be merged and displayed. If not assigned or equal to [] it will
+ * be used all instruments altogether.
+ * @param scrollType Sets scrolling to follow scoreplaying in different ways
+ * according to `ScrollType` enum values.
+ */
+export interface StaffSVGVisualizerConfig extends VisualizerConfig {
+  defaultKey?: number;
+  instruments?: number[];
+  scrollType?: ScrollType;
+}
+
+/**
+ * Displays a `NoteSecuence` as a staff on a given SVG. Staff is scaled to fit
+ * vertically `config.noteHeight` and note horizontal position can behave in
+ * two different ways: If `config.pixelsPerTimeStep` is greater than zero,
+ * horizontal position will be proportional to its starting time, allowing to
+ * pile several instances for different voices (parts). Otherwise, resulting
+ * staff will display notes in a compact form, using minimum horizontal space
+ * between musical symbols as regular paper staff does.
+ *
+ * Clef, key and time signature will be displayed at the leftmost side and the
+ * rest of the staff will scroll under this initial signature area accordingly.
+ * In case of proportional note positioning, given it starts at pixel zero, the
+ * signature area will blink meanwhile it collides with initial active notes.
+ * Key and time signature changes will be shown accordingly through score.
+ *
+ * New configuration features have been introduced through
+ * `StaffSVGVisualizerConfig` over basic `VisualizerConfig`.
+ *
+ * When connected to a player, the visualizer can also highlight
+ * the notes being currently played.
+ *
+ * You can find more info at:
+ *
+ * https://github.com/rogerpasky/staffrender-magentaviewer
+ */
+export class StaffSVGVisualizer extends BaseVisualizer {
+  private render: sr.StaffSVGRender; // The actual render.
+  private instruments: number[]; // Instruments filter to be rendered.
+  private drawnNotes: number; // Number of drawn notes. Will redraw if changed.
+
+  /**
+   * `StaffSVGVisualizer` constructor.
+   *
+   * @param sequence The `NoteSequence` to be visualized.
+   * @param div The element where the visualization should be displayed.
+   * @param config Visualization configuration options.
+   */
+  constructor(
+      sequence: INoteSequence,
+      div: HTMLDivElement,
+      config: StaffSVGVisualizerConfig = {}) {
+    super(sequence, config);
+    if ( // Overwritting super() value. Compact visualization as default.
```

Comment:
I think the indentation is still whack here. It's fine if you can't fix it, I just need to know if I need to maybe do another PR to clean it up afterwards (my vscode does it automatically)

Reply:
No worries if automatic indentation rearranges it in further PRs. I'd love to have a `CONTRIBUTING.md` file in the project with all the details about the proper contribution rules, like ending comments with period or the right VSCode configuration and plug-ins. I wasn't able to configure mine to get the automation follow your rules (didn't find the proper clang plugin).